Tests/Remora.Discord.Commands.Tests/Responders/InteractionResponderTests.cs

```diff
@@ -214,16 +215,143 @@ public async Task AreExecutedForNotFoundCommands()
                 eventMock.Setup(e => e.User).Returns(new Optional<IUser>(userMock.Object));
                 eventMock.Setup(e => e.Data).Returns(new Optional<IInteractionData>(dataMock.Object));
 
+                var result = await this.Responder.RespondAsync(eventMock.Object);
+                ResultAssert.Unsuccessful(result);  // Should return a CommandNotFoundError
+
+                // The command is never run if the InteractionResponder cannot find it via a tree search
+                // Hence no post-execution events should have run
+                _postExecutionEventMock.VerifyNoOtherCalls();
```

Comment:
A command not being found is a valid input for a post-execution event. See my comment regarding behaviour in the other file.

Reply:
Gotcha, I didn't consider that. Post-execution events are now run immediately if the `CommandService` fails to find a relevant command.

Reply:
There's also the pre-execution events to consider, so you might want to modify that as well.

Reply:
I've re-positioned the call to run pre-execution events so they once again run at an appropriate time. Unsure as to how I'd go about factoring that into the tests if that is more what you were referring to.